index.html

```diff
@@ -1080,129 +1080,155 @@
-                      message</a>. Therefore, multiple rejections SHOULD be permitted before
-                      acknowledging; allowing at least three attempts is recommended.
-                    </p>
-                  </li>
-                </ol>
-              </li>
-            </ol>
+          <li>Return <var>event</var>.
           </li>
         </ol>
-        <p>
-          When a constructor of the <a>PushEvent</a> interface, or of an interface that inherits
-          from the <a>PushEvent</a> interface, is invoked, the usual <a>steps for constructing
-          events</a> are extended to include the following steps:
-        </p>
-        <ol>
-          <li>If <var>eventInitDict</var>'s <code>data</code> member is not present, set the <code>
+        <section data-dfn-for="PushEvent">
+          <h2>
+            <dfn>PushEvent</dfn> Interface
+          </h2>
+          <pre class="idl">
+            typedef (BufferSource or USVString) PushMessageDataInit;
+
+            dictionary PushEventInit : ExtendableEventInit {
+              PushMessageDataInit data;
+            };
+
+            [Constructor(DOMString type, optional PushEventInit eventInitDict), Exposed=ServiceWorker]
+            interface PushEvent : ExtendableEvent {
+              readonly attribute PushMessageData? data;
+            };
+          </pre>
+          <p>
+            When a constructor of the <a>PushEvent</a> interface, or of an interface that inherits
+            from the <a>PushEvent</a> interface, is invoked, the usual <a>steps for constructing
+            events</a> are extended to include the following steps:
+          </p>
+          <ol>
+            <li>If <var>eventInitDict</var>'s <code>data</code> member is not present, set the <code>
             data</code> attribute of the event to <code>null</code> and terminate these steps.
-          </li>
-          <li>Set <var>b</var> to the result of <a data-lt="extract a byte sequence">extracting a
-          byte sequence</a> from the "<code>data</code>" member of <var>eventInitDict</var>.
-          </li>
-          <li>Set the <code>data</code> attribute of the event to a new <a>PushMessageData</a>
-          instance with <code>bytes</code> set to <var>b</var>.
-          </li>
-        </ol>
+            </li>
+            <li>Set <var>b</var> to the result of <a data-lt="extract a byte sequence">extracting a
+            byte sequence</a> from the "<code>data</code>" member of <var>eventInitDict</var>.
+            </li>
+            <li>Set the <code>data</code> attribute of the event to a new <a>PushMessageData</a>
+            instance with <code>bytes</code> set to <var>b</var>.
+            </li>
+          </ol>
+          <p>
+            The <dfn>data</dfn> attribute contains the data included in the <a>push message</a> when
+            included and the <a>user agent</a> verified its authenticity. The value will be set to
+            <code>null</code> in all other cases.
```

Comment:
^^^ This still is somewhat redundant (since `Receiving a Push Message` does the validation) and doesn't capture the case where a `PushEvent` is created from a `PushEventInit` structure. Would the following work for you?

> The \<dfn>data\</dfn> attribute contains the data associated with the \<a>push message\</a>. The value will be set to \<code>null\</code> when no data has been given.